src/pages/permissions/panels/EditIdentityGroupsPanel.tsx

```diff
@@ -0,0 +1,343 @@
+      ? "Group will be added to all selected identities"
+      : desiredState.groupsRemoved.has(group.name)
+        ? "Group will be removed from all selected identities"
+        : "";
+
+    return {
+      name: group.name,
+      className: "u-row",
+      columns: [
+        {
+          content: group.name,
+          role: "cell",
+          "aria-label": "Name",
+        },
+        {
+          content: group.description || "N/A",
+          role: "cell",
+          "aria-label": "Description",
+          title: group.description,
+        },
+        {
+          content: modifiedGroups.has(group.name) && (
+            <Icon name="status-in-progress-small" />
+          ),
+          role: "cell",
+          "aria-label": "Modified status",
+          className: "modified-status u-align--right",
+          title: modifiedTitle,
+        },
+      ],
+      sortData: {
+        name: group.name.toLowerCase(),
+        description: group.description.toLowerCase(),
+      },
+    };
+  });
+
+  const content = !groups.length ? (
+    <EmptyState
+      className="empty-state full-width show-left-icon"
+      image={<Icon name="plans" className="empty-state-icon" />}
+      title="No groups"
+    >
+      <p className="u-no-margin--bottom">There are no groups in the system.</p>
+      <Link to={`/ui/permissions/groups`}>
+        Create groups
+        <Icon className="external-link-icon" name="external-link" />
+      </Link>
+    </EmptyState>
+  ) : (
+    <ScrollableTable
+      dependencies={[groups, modifiedGroups.size, search, notify.notification]}
+      tableId="permission-identity-groups-table"
+      belowIds={["panel-footer"]}
+    >
+      <SelectableMainTable
+        id="permission-identity-groups-table"
+        headers={headers}
+        rows={rows}
+        sortable
+        emptyStateMsg="No groups found"
+        itemName="group"
+        parentName="server"
+        selectedNames={groupsAllocatedToAll}
+        setSelectedNames={handleModifyGroups}
+        processingNames={[]}
+        filteredNames={groups.map((group) => group.name)}
+        indeterminateNames={groupsAllocatedToSome}
+        onToggleRow={handleToggleRow}
+        hideContextualMenu
+      />
+    </ScrollableTable>
+  );
+
+  const panelTitle =
+    selectedIdentities.length > 1
+      ? `Change groups for ${selectedIdentities.length} identities`
+      : `Change groups for ${selectedIdentities[0]?.name}`;
+
+  const confirmButtonText = modifiedGroups.size
+    ? `Apply ${modifiedGroups.size} group ${pluralize("change", modifiedGroups.size)}`
+    : "Modify groups";
+
+  return (
+    <>
+      <SidePanel
+        isOverlay
+        loading={isLoading}
+        hasError={!groups}
+        className="identity-groups-panel"
+      >
+        <SidePanel.Header>
+          <SidePanel.HeaderTitle>{panelTitle}</SidePanel.HeaderTitle>
+        </SidePanel.Header>
+        <PermissionGroupsFilter onChange={setSearch} value={search} />
```

Comment:
This is a design question: To me it seems the select all is a bit too close to the search input?

![image](https://github.com/canonical/lxd-ui/assets/1155472/b2c7258f-63b2-484a-adee-3a213bdea0ac)

Reply:
Yes this is a bug, hiding the contextual menu dropdown seems to have messed up the styling a bit.